tests/sfputil_test.py

```diff
@@ -418,6 +420,197 @@ def test_show_eeprom_RJ45(self, mock_chassis):
+        a2h_upper_bytearray = bytearray([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+        expected_output = '''EEPROM hexdump for port Ethernet256 page 0h
+        A0h dump
+        00000000 03 04 07 10 00 00 00  00 00 00 00 06 67 00 00 00 |............g...|
+        00000010 08 03 00 1e 46 49 4e  49 53 41 52 20 43 4f 52 50 |....FINISAR CORP|
+        00000020 2e 20 20 20 00 00 90  65 46 54 4c 58 38 35 37 31 |.   ...eFTLX8571|
+        00000030 44 33 42 43 4c 20 20  20 41 20 20 20 03 52 00 48 |D3BCL   A   .R.H|
+        00000040 00 1a 00 00 41 55 4a  30 52 43 4a 20 20 20 20 20 |....AUJ0RCJ     |
+        00000050 20 20 20 20 31 35 31  30 32 39 20 20 68 f0 03 f6 |    151029  h...|
+        00000060 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        00000070 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        00000080 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        00000090 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        000000a0 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        000000b0 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        000000c0 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        000000d0 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        000000e0 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        000000f0 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+
+        A2h dump (Lower 128 bytes)
+        00000000 4e 00 f3 00 49 00 f8  00 90 88 71 48 8c a0 75 30 |N...I.....qH..u0|
+        00000010 19 c8 07 d0 18 9c 09  c4 27 10 09 d0 1f 07 0c 5a |........'......Z|
+        00000020 27 10 00 64 1f 07 00  9e 00 00 00 00 00 00 00 00 |'..d............|
+        00000030 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 00 |................|
+        00000040 00 00 00 00 3f 80 00  00 00 00 00 00 01 00 00 00 |....?...........|
+        00000050 01 00 00 00 01 00 00  00 01 00 00 00 00 00 00 1b |................|
+        00000060 14 02 81 b1 0d 5a 17  a5 15 87 00 00 00 00 30 00 |.....Z........0.|
+        00000070 00 00 00 00 00 00 00  00 ff ff ff ff ff ff ff 01 |................|
+
+        A2h dump (upper 128 bytes) page 0h
```

Comment:
From your testing it seems that page 2h has no output instead of something like this for page 0. Should we be consistent here for non-zero pages?

Reply:
The difference is due to the return value between page 0 v/s page 2 from the read_eeprom API. We can create a byte array for non-zero pages with all zeroes but that would also be more of modifying the result received from read_eeprom API. I think its better to leave the current output as is to keep the sanity of result received from read_eeprom API.